pkg/controller/rpaasinstance/rpaasinstance_controller.go

```diff
@@ -404,6 +434,177 @@ func (r *ReconcileRpaasInstance) reconcileNginx(nginx *nginxv1alpha1.Nginx) erro
 	return err
 }
 
+func (r *ReconcileRpaasInstance) reconcileCacheHeaterCronJob(instance *v1alpha1.RpaasInstance, plan *v1alpha1.RpaasPlan) error {
+	cronName := instance.Name + "-heater-cron-job"
+	cronJob := &batchv1beta1.CronJob{}
+
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: cronName, Namespace: instance.Namespace}, cronJob)
+	isNotFound := k8sErrors.IsNotFound(err)
+	if err != nil && !isNotFound {
+		return err
+	} else if !isNotFound {
+		logrus.Infof("CronJob %s is found, skipping creation", cronName)
+		return nil
+	}
+
+	podCommand := "rsync -avz --recursive --delete --temp-dir=/var/cache/cache-heater/temp /var/cache/nginx/rpaas/nginx /var/cache/cache-heater"
+
+	schedule := defaultCacheHeaterSchedule
+	if plan.Spec.Config.CacheHeaterSync.Schedule != "" {
+		schedule = plan.Spec.Config.CacheHeaterSync.Schedule
+	}
+
+	image := defaultCacheHeaterCronImage
+	if plan.Spec.Config.CacheHeaterSync.Image != "" {
+		image = plan.Spec.Config.CacheHeaterSync.Image
+	}
+
+	cronJob = &batchv1beta1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cronName,
+			Namespace: instance.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(instance, schema.GroupVersionKind{
+					Group:   v1alpha1.SchemeGroupVersion.Group,
+					Version: v1alpha1.SchemeGroupVersion.Version,
+					Kind:    "RpaasInstance",
+				}),
+			},
+		},
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "batch/v1beta1",
+			Kind:       "CronJob",
+		},
+		Spec: batchv1beta1.CronJobSpec{
+			Schedule: schedule,
+			JobTemplate: batchv1beta1.JobTemplateSpec{
+				Spec: batchv1.JobSpec{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							ServiceAccountName: "rpaas-cache-heater-cronjob",
+							Containers: []corev1.Container{
+								{
+									Name:  "cache-synchronize",
+									Image: image,
+									Command: []string{
+										"/bin/bash",
+									},
+									Args: []string{
+										"-c",
+										fmt.Sprintf("pods=($(kubectl -n rpaasv2-be-cme get pod -l rpaas.extensions.tsuru.io/service-name=%q -l rpaas.extensions.tsuru.io/instance-name=%q --field-selector status.phase=Running -o=jsonpath='{.items[*].metadata.name}')); ", instance.Namespace, instance.Name) +
+											"for pod in ${pods[@]}; do " +
+											fmt.Sprintf("kubectl -n %q exec ${pod} -- %q;", instance.Namespace, podCommand) +
+											"if [[ $? == 0 ]]; then exit 0; fi; " +
+											"done",
+									},
```

Comment:
Instead of interpolating the args here I think we could have a single static `defaultCacheHeaterCmds` and rely on injected Env variables for service name and instance. This would make this more flexible for custom images or custom commands in the future. It could be defined as:

```go
var defaultCacheHeaterCmds = []string{
	"/bin/bash",
	"-c",
	`
pods=($(kubectl -n rpaasv2-be-cme get pod -l rpaas.extensions.tsuru.io/service-name=${SERVICE_NAME} -l rpaas.extensions.tsuru.io/instance-name=${INSTANCE_NAME} --field-selector status.phase=Running -o=jsonpath='{.items[*].metadata.name}'));
for pod in ${pods[@]}; do
	kubectl -n ${SERVICE_NAME} exec ${pod} -- ${POD_CMD};
	if [[ $? == 0 ]]; then 
		exit 0;
	fi
done
`,
}
```

And in the container spec:

```go
Env: []corev1.EnvVar{
	{Name: "SERVICE_NAME", Value: instance.Namespace},
	{Name: "INSTANCE_NAME", Value: instance.Name},
	{Name: "POD_CMD", Value: podCommand},
}
```